installer.sh

```diff
@@ -360,37 +360,48 @@ else
   CHANGEPASSWORD="y"
 fi
 
-if [ "$CHANGEPASSWORD" == "y" ]; then
-  unset CHARCOUNT
-  echo -n "Set the password to access the Dashboard: "
-  CHARCOUNT=0
+
+read_password() {
+  local CHARCOUNT=0
+  local PASSWORD=""
   while IFS= read -p "$PROMPT" -r -s -n 1 CHAR
   do
     # Enter - accept password
     if [[ $CHAR == $'\0' ]] ; then
-      if [ $CHARCOUNT -gt 0 ] ; then # Make sure password character length is greater than 0.
-        break
-      else
-        echo
-        echo -n "Invalid password input. Enter a password with character length greater than 0:"
-        continue
-      fi
+      break
     fi
     # Backspace
     if [[ $CHAR == $'\177' ]] ; then
       if [ $CHARCOUNT -gt 0 ] ; then
         CHARCOUNT=$((CHARCOUNT-1))
         PROMPT=$'\b \b'
-        DASHPASS="${DASHPASS%?}"
+        PASSWORD="${PASSWORD%?}"
       else
         PROMPT=''
       fi
     else
       CHARCOUNT=$((CHARCOUNT+1))
       PROMPT='*'
-      DASHPASS+="$CHAR"
+      PASSWORD+="$CHAR"
     fi
   done
+  echo $PASSWORD
+}
+
+if [ "$CHANGEPASSWORD" = "y" ]; then
+  valid_pass=false
+  while [ "$valid_pass" = false ] ;
+  do
+      echo -n -e "Password requirements: min 8 characters, at least 1 letter, at least 1 number, at least 1 special character \nSet the password to access the Dashboard:"
+      DASHPASS=$(read_password)
```

Comment:
display the list of valid special characters here.

Reply:
![image](https://github.com/user-attachments/assets/de3e5dd1-7f01-49d3-a189-db98fbd89447)